tests/unit/s2n_ktls_test.c

```diff
@@ -38,5 +75,163 @@ int main(int argc, char **argv)
         EXPECT_FALSE(cipher.ktls_supported);
     };
 
+    if (!s2n_ktls_is_supported_on_platform()) {
+        DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+                s2n_connection_ptr_free);
+        int fd = 1;
+        EXPECT_OK(s2n_test_configure_mock_ktls_connection(server_conn, fd, true));
+
+        EXPECT_FAILURE_WITH_ERRNO(s2n_connection_ktls_enable_send(server_conn), S2N_ERR_KTLS_UNSUPPORTED_PLATFORM);
+        EXPECT_FAILURE_WITH_ERRNO(s2n_connection_ktls_enable_recv(server_conn), S2N_ERR_KTLS_UNSUPPORTED_PLATFORM);
+    } else {
+        /* ktls handshake must be complete */
+        {
```

Comment:
Rather than putting all your tests in this if, it might be cleaner to go:
```suggestion
    if (!s2n_ktls_is_supported_on_platform()) {
        DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
                s2n_connection_ptr_free);
                
        int fd = 1;
        EXPECT_OK(s2n_test_configure_mock_ktls_connection(server_conn, fd, true));

        EXPECT_FAILURE_WITH_ERRNO(s2n_connection_ktls_enable_send(server_conn), S2N_ERR_KTLS_UNSUPPORTED_PLATFORM);
        EXPECT_FAILURE_WITH_ERRNO(s2n_connection_ktls_enable_recv(server_conn), S2N_ERR_KTLS_UNSUPPORTED_PLATFORM);
        END_TEST();
    }
    
   /* ktls handshake must be complete */
   {
```
I'd also move the "ktls not supported" test to the very top of the file, before the cipher ones.